### ISSUE TITLE
[question] Softmax and crossentropy for image tensors

### ISSUE THREAD

**Issue by fferroni** [fchollet/keras#7490]
Using keras 2.0.4 and tf 1.2.0

I have seen multiple differing answers with regards to using softmax and crossentropy loss for image tensors, for the purposes of pixel-level classification.
I would like to know if at the moment the softmax activation is capable of being used with image tensors (i.e. last dimension in tensorflow is the channels, one for each class) and if we can use categorical_crossentropy also out of the box. i.e.
```
model.add(Convolution2D(nb_labels, (1, 1), border_mode='valid', activation='softmax')
model.compile(loss="categorical_crossentropy", ...)
```

Looking at the source code it seems like it would work (at least for tensorflow backend), since the axis for these operations defaults to -1. But I would like to check if this gives correct results.

However, I have seen answers which suggest doing the following:
```
model.add(Convolution2D(nb_labels, (1, 1), border_mode='valid'))
model.add(Reshape((nb_labels, img_h * img_w)))
model.add(Permute((2, 1)))
model.add(Activation('softmax'))
model.compile(loss="categorical_crossentropy", ...)
```

Thanks!
